### PR TITLE
Mileston 3

### PR DESCRIPTION
**Pull request by AbigiyaTY**
In this milestone I:

- Used Redux in React components
- Used Redux store to display books
- User can Add and Remove Books

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -1,23 +1,20 @@
...
+export const booksSlice = createSlice({
+  name: 'books',
+  initialState: { value: books },
+  reducers: {
+    addBook: (state, action) => {
+      state.value.push(action.payload);
+    },
+
+    deleteBook: (state, action) => {
+      // eslint-disable-next-line no-param-reassign
```
Comment (FrederickMih) [inline src/redux/books/books.js]: - [ ] Please do not disable Linters. They monitor your code quality and best practices and flag coding errors. Fix these errors.

File: src/Components/AddBook/AddBook.js [hunk 2]
```diff
@@ -1,22 +1,75 @@
...
+                addBook({
+                  id: uuid(),
+                  title,
+                  author,
+                }),
+              );
+            }}
+          >
+            Add Book
+          </button>
```
Comment (FrederickMih) [inline src/Components/AddBook/AddBook.js]: - Great job for having this button to add a book :+1: 

- [ ] Based on the requirements of this project, this button should **use** a function `submitBookToStore` (name can be different) made in this component (the name may be different), Hence you'll implement this function up :point_up: and use it inside this button.

File: src/Components/AddBook/AddBook.js [hunk 3]
```diff
@@ -1,22 +1,75 @@
...
+                <h3 className="col-12 author">{book.author}</h3>
+              </div>
+              <button
+                type="button"
+                className="col-3 my-2 removeButton"
+                onClick={() => {
+                  dispatch(deleteBook({ id: book.id }));
+                }}
+              >
+                Remove
```
Comment (FrederickMih) [inline src/Components/AddBook/AddBook.js]: - [ ] Based on the requirements of this project, this button should be found in another component called `Book`(We need to learn how to make use of several components in our app :+1: ). This button should use a function `removeBookFromStore` (name can be different) made in the component(We needs to learn how to craft a function and make use of it :+1: ). Kindly fix :wrench: this

*

File: src/Components/Categories/Categories.js [hunk 4]
```diff
@@ -1,13 +1,24 @@
...
+        <hr className="col-12" />
+        <button
+          type="button"
+          className="col-4  my-2 py-5 px-3 catButton "
+          onClick={() => {
+            dispatch(categoryActionReducer());
+          }}
+        >
+          Check status
+        </button>
```
Comment (FrederickMih) [inline src/Components/Categories/Categories.js]: - [ ] Once more, based on the requirements of this project, this button should **use** a function `checkCategoriesStatus` (name can be different) made in the component. So you'll need to implement a function up :point_up: inside this component and use it inside this button

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @AbigiyaTY , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




## Highlights

- Correct GitHub flow :heavy_check_mark: 
- PR has a title and summary :heavy_check_mark: 
- No Linter error :heavy_check_mark: 

## Required Changes ♻️

- [ ]  Please kindly fix the console errors on the screenshot below:


Screenshot from 2022-10-07 19-46-46


_Check the comments under the review._

## Optional suggestions 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by descholar-ceo — APPROVED** [review]
Hi @AbigiyaTY 🤝 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
👍🏽 Addressed all of the requested changes ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**